centreon/src/Core/HostGroup/Application/UseCase/UpdateHostGroup/UpdateHostGroup.php

```diff
@@ -0,0 +1,246 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\HostGroup\Application\UseCase\UpdateHostGroup;
+
+use Assert\AssertionFailedException;
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ConflictResponse;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\NoContentResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\Domain\Common\GeoCoords;
+use Core\Domain\Exception\InvalidGeoCoordException;
+use Core\HostGroup\Application\Exceptions\HostGroupException;
+use Core\HostGroup\Application\Repository\ReadHostGroupRepositoryInterface;
+use Core\HostGroup\Application\Repository\WriteHostGroupRepositoryInterface;
+use Core\HostGroup\Domain\Model\HostGroup;
+use Core\Infrastructure\Common\Api\DefaultPresenter;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+use Core\ViewImg\Application\Repository\ReadViewImgRepositoryInterface;
+
+final class UpdateHostGroup
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly ReadHostGroupRepositoryInterface $readHostGroupRepository,
+        private readonly WriteHostGroupRepositoryInterface $writeHostGroupRepository,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+        private readonly ReadViewImgRepositoryInterface $readViewImgRepository,
+        private readonly ContactInterface $contact
+    ) {
+    }
+
+    /**
+     * @param int $hostGroupId
+     * @param UpdateHostGroupRequest $request
+     * @param DefaultPresenter $presenter
+     */
+    public function __invoke(
+        int $hostGroupId,
+        UpdateHostGroupRequest $request,
+        PresenterInterface $presenter
+    ): void {
+        try {
+            if ($this->contact->isAdmin()) {
+                $response = $this->updateHostGroupAsAdmin($hostGroupId, $request);
+            } elseif ($this->contactCanPerformWriteOperations()) {
+                $response = $this->updateHostGroupAsContact($hostGroupId, $request);
+            } else {
+                $response = new ForbiddenResponse(HostGroupException::accessNotAllowedForWriting());
+            }
+
+            if ($response instanceof NoContentResponse) {
+                $presenter->present($response);
+                $this->info('Update host group', ['request' => $request]);
```

Comment:
```suggestion
                $this->info('Update host group', ['request' => $request]);
                return ;
```

Reply:
To me : not worth. It does not improve readibility.